tests/Concerns/SkipsOnErrorTest.php

```diff
@@ -113,4 +118,126 @@ public function model(array $row)
+                ]);
+            }
+
+            /**
+             * @return array
+             */
+            public function rules(): array
+            {
+                return [
+                    '1' => Rule::in(['patrick@maatwebsite.nl']),
+                ];
+            }
+
+            /**
+             * @param  Throwable  $e
+             */
+            public function onError(Throwable $e)
+            {
+                Assert::assertInstanceOf(ValidationException::class, $e);
+                Assert::stringContains($e->getMessage(), 'The selected 1 is invalid');
+
+                $this->errors++;
+            }
+        };
+
+        $import->import('import-users.xlsx');
+
+        $this->assertEquals(1, $import->errors);
+        $this->assertEquals(1, $import->processedRows); // Only the valid row should be processed
+
+        // Should have inserted the valid row
+        $this->assertDatabaseHas('users', [
+            'email' => 'patrick@maatwebsite.nl',
+        ]);
+
+        // Should have skipped inserting the invalid row
+        $this->assertDatabaseMissing('users', [
+            'email' => 'taylor@laravel.com',
+        ]);
+    }
+
+    public function test_can_skip_errors_and_collect_all_errors_when_using_oneachrow_with_validation()
+    {
+        $import = new class implements OnEachRow, WithValidation, SkipsOnError
+        {
+            use Importable, SkipsErrors;
+
+            public $processedRows = 0;
+
+            /**
+             * @param  Row  $row
+             */
+            public function onRow(Row $row)
+            {
+                $this->processedRows++;
+
+                // This will be called for valid rows
+                $rowArray = $row->toArray();
+
+                User::create([
+                    'name'     => $rowArray[0],
+                    'email'    => $rowArray[1],
+                    'password' => 'secret',
+                ]);
+            }
+
+            /**
+             * @return array
+             */
+            public function rules(): array
+            {
+                return [
+                    '1' => Rule::in(['patrick@maatwebsite.nl']),
+                ];
+            }
+        };
+
+        $import->import('import-users.xlsx');
+
+        $this->assertCount(1, $import->errors());
+        $this->assertEquals(1, $import->processedRows); // Only the valid row should be processed
+
+        /** @var Throwable $e */
+        $e = $import->errors()->first();
+
+        $this->assertInstanceOf(ValidationException::class, $e);
+        $this->stringContains($e->getMessage(), 'The selected 1 is invalid');
+
+        // Should have inserted the valid row
+        $this->assertDatabaseHas('users', [
+            'email' => 'patrick@maatwebsite.nl',
+        ]);
+
+        // Should have skipped inserting the invalid row
+        $this->assertDatabaseMissing('users', [
+            'email' => 'taylor@laravel.com',
+        ]);
+    }
```

Comment:
Here we should also have a test with `new class implements OnEachRow, SkipsOnError`

Reply:
Completed.